kubeflow/trainer/rhai/speculator.py

```
@@ -0,0 +1,1388 @@
+
+        regen_dataset_map = {
+            "sharegpt": "magpie",
+            "magpie": "magpie",
+            "ultrachat": "ultrachat",
+            "gsm8k": "gsm8k",
+        }
+        regen_dataset = regen_dataset_map.get(dataset_name, "magpie")
+        regen_output = str(Path(save_path) / "regenerated_responses.jsonl")
+        chat_endpoint = endpoint.rstrip("/").rsplit("/v1", 1)[0] + "/v1/chat/completions"
+
+        regen_script_path = "/tmp/response_regeneration.py"
```

Ruff (0.15.21)

[error] Probable insecure usage of temporary file or directory: "/tmp/response_regeneration.py"

(S108)